client/transactions/filters/config.ts

```diff
@@ -52,169 +52,228 @@ const loanSelectionOptions = loanDefinitions.map( ( loanDefinition ) => {
-	title: __(
-		'Transactions match {{select /}} filters',
-		'woocommerce-payments'
-	),
-	filters: {
-		date: {
-			labels: {
-				add: __( 'Date', 'woocommerce-payments' ),
-				remove: __(
-					'Remove transaction date filter',
-					'woocommerce-payments'
-				),
-				rule: __(
-					'Select a transaction date filter match',
-					'woocommerce-payments'
-				),
-				/* translators: A sentence describing a Transaction date filter. */
-				title: __(
-					'{{title}}Date{{/title}} {{rule /}} {{filter /}}',
-					'woocommerce-payments'
-				),
-				filter: __(
-					'Select a transaction date',
-					'woocommerce-payments'
-				),
-			},
-			rules: [
+export const getFilters = (
+	depositCurrencyOptions: TransactionsFilterEntryType[],
+	showDepositCurrencyFilter: boolean
+): [ TransactionsFilterType, TransactionsFilterType ] => {
+	return [
+		{
+			label: __( 'Deposit currency', 'woocommerce-payments' ),
+			param: 'store_currency_is',
+			staticParams: [
+				'paged',
+				'per_page',
+				'orderby',
+				'order',
+				'search',
+				'filter',
+				'type_is',
+				'type_is_not',
+				'date_before',
+				'date_after',
+				'date_between',
+			],
+			showFilters: () => showDepositCurrencyFilter,
+			filters: [
 				{
-					value: 'before',
-					label: __( 'Before', 'woocommerce-payments' ),
+					label: __( 'All currencies', 'woocommerce-payments' ),
+					value: '---',
 				},
+				...depositCurrencyOptions,
+			],
+			defaultValue: '---',
+		},
+		{
+			label: __( 'Show', 'woocommerce-payments' ),
+			param: 'filter',
+			staticParams: [
+				'paged',
+				'per_page',
+				'orderby',
+				'order',
+				'search',
+				'store_currency_is',
+			],
+			showFilters: () => true,
+			filters: [
 				{
-					value: 'after',
-					label: __( 'After', 'woocommerce-payments' ),
+					label: __( 'All transactions', 'woocommerce-payments' ),
+					value: 'all',
 				},
 				{
-					value: 'between',
-					label: __( 'Between', 'woocommerce-payments' ),
+					label: __( 'Advanced filters', 'woocommerce-payments' ),
+					value: 'advanced',
 				},
 			],
-			input: {
-				component: 'Date',
-			},
 		},
-		type: {
-			labels: {
-				add: __( 'Type', 'woocommerce-payments' ),
-				remove: __(
-					'Remove transaction type filter',
-					'woocommerce-payments'
-				),
-				rule: __(
-					'Select a transaction type filter match',
-					'woocommerce-payments'
-				),
-				/* translators: A sentence describing a Transaction type filter. */
-				title: __(
-					'{{title}}Type{{/title}} {{rule /}} {{filter /}}',
-					'woocommerce-payments'
-				),
-				filter: __(
-					'Select a transaction type',
-					'woocommerce-payments'
-				),
-			},
-			rules: [
-				{
-					value: 'is',
-					/* translators: Sentence fragment, logical, "Is" refers to searching for transactions matching a chosen transaction type. */
-					label: _x(
-						'Is',
-						'transaction type',
+	];
+};
+
+/**
+ * TODO: Add an interface here for advanced filters.
```

Comment:
It could also be solved by adjusting `explicit-module-boundary-types` rule, letting us use TS inferred type. I think we can mention that for future TypeScript work. Something like `... or adjust ESLint rules to allow using inferred type`. More info here: paJDYF-4tS-p2#comment-12406

Reply:
Thanks! Added a comment in 1ee4f397. I think this is probably one that's better to wait for the planned work to initially define some types (or used inferred types) across the board.